src/main/java/at/ac/uibk/dps/cirrina/classes/transition/TransitionClass.java

```diff
@@ -80,10 +82,10 @@ public void accept(PlantUmlVisitor visitor) {
    * @return True if the transition can be taken based on the guards, otherwise false.
    * @throws UnsupportedOperationException If the transition could not be evaluated.
    */
-  public boolean evaluate(Extent extent) throws UnsupportedOperationException {
+  public boolean evaluate(Extent extent, TracingAttributes tracingAttributes, Span span) throws UnsupportedOperationException {
```

Comment:
As discussed, please, if there is a way to make this somewhat nicer from a function call perspective (not having to pass extra arguments), try it. Otherwise, we can keep it like this